rfcs/0024-incremental-ledger-catchup.md

```diff
@@ -0,0 +1,81 @@
+## Terminology
+
+history root: the oldest node in root history
+
+## Summary
+
+The origin transition catchup design has performance issues. In
+current implementation, we would always downloading a list of transitions
+from the requested hash to peer's history root.
+The minimum number of transitions that transition catchup would request is
+k + 1 under current implementation. This is not very efficient in a realistic
+setting where k is ~3000. The new design features
```

Comment:
It would be nice to know at what value of `k` the distributed design offers better performance